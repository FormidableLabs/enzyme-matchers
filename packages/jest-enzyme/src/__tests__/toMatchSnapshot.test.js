const React = require('react');
const { shallow, mount } = require('enzyme');

describe('toMatchSnapshot OVERIDE', () => {
  [shallow, mount].forEach(builder => {
    it(`works for enzyme objects (${builder.name})`, () => {
      class Fixture extends React.Component {
        onClick() {}
        render() {
          return (
            <div onClick={this.onClick}>
              <span>foobar</span>
            </div>
          );
        }
      }
      const wrapper = builder(<Fixture />);

      expect(wrapper).toMatchSnapshot();
    });
  });


  it('falls back to the default for non enzyme objects', () => {
    expect(1).toMatchSnapshot();
    expect({nodes: []}).toMatchSnapshot();
  });
});
