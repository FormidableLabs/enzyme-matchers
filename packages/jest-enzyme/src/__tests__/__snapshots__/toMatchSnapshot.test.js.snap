exports[`toMatchSnapshot OVERIDE falls back to the default for non enzyme objects 1`] = `1`;

exports[`toMatchSnapshot OVERIDE falls back to the default for non enzyme objects 2`] = `
Object {
  "nodes": Array [],
}
`;

exports[`toMatchSnapshot OVERIDE works for enzyme objects (mount) 1`] = `
<Fixture>
  <div
    onClick={[Function]}>
    <span>
      foobar
    </span>
  </div>
</Fixture>
`;

exports[`toMatchSnapshot OVERIDE works for enzyme objects (shallow) 1`] = `
<div
  onClick={[Function]}>
  <span>
    foobar
  </span>
</div>
`;
